dump/src/reader/v4/mod.rs

```diff
@@ -147,16 +149,31 @@ pub struct V4IndexReader {
 }
 
 impl V4IndexReader {
-    pub fn new(name: String, path: &Path) -> Result<Self> {
+    pub fn new(name: String, path: &Path, tasks: BufReader<File>) -> Result<Self> {
         let meta = File::open(path.join("meta.json"))?;
         let meta: DumpMeta = serde_json::from_reader(meta)?;
 
+        let mut index_tasks: Vec<Task> = vec![];
+
+        for line in tasks.lines() {
+            let task: Task = serde_json::from_str(&line?)?;
+
+            if task.index_uid.to_string() == name {
+                index_tasks.push(task)
+            }
+        }
```

Comment:
Could you compute the `created_at` and `updated_at` here directly to avoid allocating a `Vec` with all tasks and avoid scanning the whole tasks queue if there is only a few tasks concerning the index.

```suggestion
        let created_at = None;
        let updated_at = None;

        for line in tasks.lines() {
            let task: Task = serde_json::from_str(&line?)?;

            if task.index_uid == &name {
                if updated_at.is_none() {
                    updated_at = Some(...); // get the timestamp of the first task matching our index name
                }
                if task.type == Created { // find a way to check if the task type was an IndexCreation
                    created_at = Some(...); // get the timestamp of the task creation.
                    break; // we don't need to read any more task and can exit.
                }
            }
        }
```

Reply:
Thanks for implementing the changes @funilrys :tada: